commons/src/main/java/org/infinispan/commons/equivalence/EntryEquivalence.java

```diff
@@ -0,0 +1,62 @@
+package org.infinispan.commons.equivalence;
+
+import java.util.Map;
+
+/**
+ * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+ */
+public class EntryEquivalence<K, V> implements Equivalence<Map.Entry<K, V>> {
```

Comment:
I still think the entry equivalence isn't necessary... the `Collection.retainAll()` javadoc says "Retains only the elements in this collection that are contained in the specified collection", so it's defined based on the collection parameter's `contains()`, not on our `contains()`.